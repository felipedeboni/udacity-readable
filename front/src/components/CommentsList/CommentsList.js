import React from 'react';
import PropTypes from 'prop-types';
import sortBy from 'sort-by';
import { CommentListItem } from 'components';
import { CommentFormContainer, CommentContainer } from 'containers';
import './CommentsList.css';

const CommentsList = ({ postId, comments, category }) => {
  const sortedComments = comments.sort(sortBy('-timestamp'));
  const commentsCount = sortedComments.length;
  return (
    <section className="comments">
      <div className="comments__counter">
        {commentsCount === 0 && 'No comments'}
        {commentsCount > 0 && `${commentsCount} comment`}
        {commentsCount > 1 && 's'}
      </div>
      <ul className="comments__list list-unstyled">
        <li className="comments__form">
          <CommentFormContainer postId={postId} />
        </li>
        {sortedComments.map(comment => (
          <CommentContainer
            key={comment.id}
            comment={comment}
            component={CommentListItem}
          />
        ))}
      </ul>
    </section>
  );
};

CommentsList.propTypes = {
  postId: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  comments: PropTypes.array.isRequired
};

export default CommentsList;
