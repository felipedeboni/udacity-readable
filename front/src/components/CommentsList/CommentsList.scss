.comments {
  border-top: 1px solid #F3F3F3;
  margin: 30px 0 0;

  &__counter {
    font-weight: bold;
    padding: 10px 0;
  }

  &__list {
    padding: 15px;
  }

  &__form {
    padding-bottom: 30px;
    border-bottom: 2px solid #f3f3f3;
  }
}
