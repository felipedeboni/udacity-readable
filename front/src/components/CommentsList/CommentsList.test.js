import React from 'react';
import { shallow } from 'enzyme';
import CommentsList from './CommentsList';

describe('CommentsList', () => {
  it('renders without crashing', () => {
    const { comments } = global.mock;

    const test = (
      <CommentsList category="react" comments={comments} postId="1" />
    );

    expect(shallow(test)).toMatchSnapshot();
  });
});
