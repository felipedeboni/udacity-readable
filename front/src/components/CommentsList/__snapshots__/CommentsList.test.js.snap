// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsList renders without crashing 1`] = `
<section
  className="comments"
>
  <div
    className="comments__counter"
  >
    2 comment
    s
  </div>
  <ul
    className="comments__list list-unstyled"
  >
    <li
      className="comments__form"
    >
      <Connect(CommentFormContainer)
        postId="1"
      />
    </li>
    <Connect(CommentContainer)
      comment={
        Object {
          "author": "thingone",
          "body": "Comments. Are. Cool.",
          "deleted": false,
          "id": "8tu4bsun805n8un48ve89",
          "parentDeleted": false,
          "parentId": "8xf0y6ziyjabvozdd253nd",
          "timestamp": 1469479767190,
          "voteScore": -5,
        }
      }
      component={[Function]}
      key="8tu4bsun805n8un48ve89"
    />
    <Connect(CommentContainer)
      comment={
        Object {
          "author": "thingtwo",
          "body": "Hi there! I am a COMMENT.",
          "deleted": false,
          "id": "894tuq4ut84ut8v4t8wun89g",
          "parentDeleted": false,
          "parentId": "8xf0y6ziyjabvozdd253nd",
          "timestamp": 1468166872634,
          "voteScore": 6,
        }
      }
      component={[Function]}
      key="894tuq4ut84ut8v4t8wun89g"
    />
  </ul>
</section>
`;
